service/raft/raft_group0.cc

```diff
@@ -929,9 +929,14 @@ future<> raft_group0::modify_raft_voter_status(
 
```

Comment:
Would be good to extend the comment inside .hh to specify this behavior (i.e. doesn't affect non-members)

Although, unfortunately, it's not guaranteed, because the check + modify_config they are not atomic :( Nothing like group 0 guard is involved here. So between the check and the command getting appended to the log, membership could change.

So we could say that the behavior is only guaranteed if no concurrent membership changes are happening.

I wonder if in our context this is indeed the case

Reply:
> Would be good to extend the comment inside .hh to specify this behavior (i.e. doesn't affect non-members)

sure

> I wonder if in our context this is indeed the case

Probably not. A removenode initiator does this config change. I don't see why another node wouldn't do another config change at this point, for example, while initiating another removenode.

I also need to check if this commit doesn't break anything in the voter set PR.